### ISSUE TITLE
Over-saturated problem in distillaiton

### ISSUE THREAD

**Issue by phyllispeng123** [tianweiy/CausVid#18]
Hi @tianweiy, I use the DMD distillation method provided by CausVid in Wan2.1 , and meet over-saturated problem in >=8-step distillation, while loss of detail problem in portraits in 2-step distillation. I wonder if you have met the same over-saturated issue? do you have any recommendation on more-step distillation ? many thanks !!

**Comment by tianweiy** [comment]
thanks for the interest. I don't have much clue. But one thing you can try is to also use the ode loss during dmd training (basically combine with this script https://github.com/tianweiy/CausVid/blob/master/causvid/train_ode.py)

I remember adding gan loss might also help but last year when I tried, the motion seems worse so I didn't fully investigate. 

people told me this extension of dmd might be helpful for addressing oversaturating issues too but it seems hard to implement and I haven't got chance to try. https://arxiv.org/abs/2406.04103

**Comment by hkunzhe** [comment]
@phyllispeng123 I met the same problem. What is your settings (denoising_step_list and timestep_shift)? Have you figured it out?

**Comment by szj0910** [comment]
@phyllispeng123 @xunhuang1995 hi, Have you solved this problem yet？I met the same problem, Wan1.3b Bidirectional DMD Training using default configuration, the generated video has a particularly high saturation level

**Comment by szj0910** [comment]
@tianweiy Could you please confirm whether the wan2.1 bidirectional_checkpoint model provided in the Git repository was trained on the MixKit Dataset (6K videos)? I have been unable to reproduce the model's performance—the distilled model I trained generates videos with excessively high saturation. I used the MixKit Dataset (6K videos) along with the default configuration files.

I suspect that the open-sourced bidirectional_checkpoint model might have been trained on additional data, or perhaps there are some key details in the training process that I overlooked. I would greatly appreciate your insights on this matter. Looking forward to your response!

**Comment by hkunzhe** [comment]
@tianweiy Hi, Any insights or suggestions to tune the timestep shift, guidance scale, or denoising steps https://github.com/tianweiy/CausVid/issues/10？I reproduce the 1.3B 3steps model with the default setting. The color of the video generated by the model is relatively normal. However, when I increase the steps to 4, whether close to x_t (larger noise) or x_0 (smaller noise), the generated video tends to have issues with excessive saturation, which is more noticeable on the 14B model.

**Comment by tianweiy** [comment]
@hkunzhe  I don't have much experiences with Wan. 

> Any insights or suggestions to tune the timestep shift, guidance scale, or denoising steps

I think tilmestep shift seems already large enough. Larger guidance scale will lead to more overexposure, Maybe try the ODE or GAN loss. I feel they should work ...

**Comment by thbupt** [comment]
> a few general comments on the overexposure problem:
> 
> So we noticed a little bit overexposure with our internal model's results shown here https://causvid.github.io but it is much less severe than the wan's one implemented in that repo. I unfortunately don't know why. A few differences are
> 
> 1. we use different architecture and the internal teacher model is trained for a long time on some other dataset.
> 2. the distillation dataset is also different.
> 
> Therefore, I guess wan model might need some other training strategies than the current one to get perfect results. For instance,
> 
> 1. maybe we should also run the train_ode loss to pretrain the bidirectional model  (https://github.com/tianweiy/CausVid/blob/master/causvid/train_ode.py)
> 2. maybe we should do joint ode and dmd loss in the dmd stage like DMD 1
> 3. maybe we need to add a gan loss (in my prior experiment, this can work for reducing overexposure). The gan doesn't have to be using the full fake network as discriminator. Just adding a branch to the frozen real diffusion model like SD3 works better actually.
> 4. doing the backward_simulation seems a little bit better (we never do it in the paper and with internal model, it seems not needed but anyway...)
> 5. and probably a better dataset might be helpful.

@tianweiy I met similar overexposure problem. Regarding the second suggestion you mentioned, which involves adopting the ODE loss, it appears to require additional use of the dataset employed in ODE initialization. Since DMD training utilizes ground truth latent, how about to simply add a denoised loss to the generator? Thank you for your advice.

**Comment by thbupt** [comment]
@phyllispeng123 @xunhuang1995 @szj0910 Have you solved the overexposure problem? In my experiment, this problem is not alleviated even adding ode loss(with ode loss weight = 0.25)
